src/components/pagination/pagination.module.css

```diff
@@ -0,0 +1,182 @@
+.pagination {
+	--token-pagination-control-height: 36px;
+	--token-pagination-control-padding-horizontal: 11px;
+	--token-pagination-indicator-height: 2px;
+	--token-pagination-nav-control-focus-inset: 4px;
+
+	composes: hds-typography-body-100 from global;
+	font-weight: var(--token-typography-font-weight-medium);
+	display: grid;
+	grid-template-areas: 'left center right';
+	grid-template-rows: auto;
+	grid-template-columns: 1fr auto 1fr;
+	align-items: center;
+	margin: 0 auto;
+	color: var(--token-color-foreground-primary);
+}
+
+/* <Info> */
+.info {
+	grid-area: left;
+	justify-self: flex-start;
+	white-space: nowrap;
+	composes: hds-typography-body-100 from global;
+	font-weight: var(--token-typography-font-weight-medium);
+}
+
+/* <Nav> */
+.nav {
+	display: flex;
+	grid-area: center;
+	composes: hds-typography-body-100 from global;
+	font-weight: var(--token-typography-font-weight-medium);
+}
+
+.page-list {
+	display: flex;
+	margin: 0;
+	padding: 0;
+}
+
+.page-item {
+	position: relative;
+	list-style-type: none;
+}
+
+.is-selected {
+	color: var(--token-color-foreground-action);
+	&:focus-within {
+		color: var(--token-color-foreground-action-active);
+	}
+
+	/* small underline */
+	&::before {
+		position: absolute;
+		right: 6px;
+		bottom: 0;
+		left: 6px;
+		height: var(--token-pagination-indicator-height);
+		margin: 0 auto;
+		background-color: currentColor;
+		border-radius: 2px;
+		content: '';
+	}
+}
+
+.control {
+	color: inherit;
+	border-radius: 5px;
+	font-weight: var(--token-typography-font-weight-medium);
+	text-decoration: none;
+	background-color: transparent;
+	border: none;
+	position: relative;
+	outline: transparent;
+
+	&:hover {
+		color: var(--token-color-foreground-action-hover);
+	}
+
+	&:disabled {
+		color: var(--token-color-foreground-disabled);
+		cursor: not-allowed;
+	}
+
+	/* focus ring */
+	&::before {
+		top: var(--token-pagination-nav-control-focus-inset);
+		right: var(--token-pagination-nav-control-focus-inset);
+		bottom: var(--token-pagination-nav-control-focus-inset);
+		left: var(--token-pagination-nav-control-focus-inset);
+		z-index: -1;
+		border-radius: 5px;
+		position: absolute;
+		content: '';
+	}
+
+	/* tab focus or click focus */
+	&:focus {
+		color: var(--token-color-foreground-action);
+		&::before {
+			box-shadow: var(--token-focus-ring-action-box-shadow);
+		}
+	}
```

Comment:
🙋🏻 Is it possible to use the [`.g-focus-ring-from-box-shadow` helper](https://github.com/hashicorp/dev-portal/blob/main/src/pages/style.css#L176-L185) here to decrease the maintenance effort of the shared style? It avoids using `:focus` since that adds an unwanted ring after clicking, also handles setting `outline: transparent` so that the focus ring works in high contrast modes.

Reply:
@ashleemboyer done in https://github.com/hashicorp/dev-portal/pull/1508/commits/02005095f2c1c6be25531993bae9560ac5c5dcd8